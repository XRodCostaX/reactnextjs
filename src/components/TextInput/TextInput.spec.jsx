import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { describe, vi } from 'vitest';
import { TextInput } from '.';

describe('<TextInput />', () => {
  it('should have a value of searchValue', () => {
    const fn = vi.fn();

    const { debug } = render(<TextInput handleChange={fn} searchValue="testando" />);
    const input = screen.getByPlaceholderText(/type your search/i);
    expect(input).toBeInTheDocument();
    expect(input.value).toBe('testando');
    debug();
  });
  it('should call a handleChange function on each key pressed', async () => {
    const fn = vi.fn();

    render(<TextInput handleChange={fn} />);
    const input = screen.getByPlaceholderText(/Type your Search/i);
    const user = userEvent.setup();
    const value = 'o valor';

    await user.type(input, value);
    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapshot', () => {
    const fn = vi.fn();

    const { container } = render(<TextInput handleChange={fn} searchValue="" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
