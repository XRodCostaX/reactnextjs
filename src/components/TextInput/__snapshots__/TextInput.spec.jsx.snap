// Vitest Snapshot v1

exports[`<TextInput /> > should match snapshot 1`] = `
<div
  class="sc-bcXHqe fokZYD"
>
  <input
    class="text-input"
    placeholder="Type your Search"
    type="search"
    value=""
  />
</div>
`;
